catalog/transformer/catalog-transformer-streaming-impl/src/main/java/org/codice/ddf/transformer/xml/streaming/impl/XMLSaxEventHandlerImpl.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.transformer.xml.streaming.impl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.codice.ddf.transformer.xml.streaming.SaxEventHandler;
+import org.xml.sax.Attributes;
+import org.xml.sax.Locator;
+import org.xml.sax.SAXException;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.AttributeImpl;
+
+/**
+ * A sax event handler used to parse urn:catalog:metacard {@link Metacard}s. By default, handles all elements defined in the {@link XMLSaxEventHandlerImpl#xmlToMetacard}
+ * These defaults can be overridden by passing a different {@link Map} in {@link XMLSaxEventHandlerImpl#setXmlToMetacard(Map)}
+ */
+public class XMLSaxEventHandlerImpl implements SaxEventHandler {
+
+    private List<Attribute> attributes;
+
+    private String elementBeingRead;
+
+    private StringBuilder textDataOfElement;
+
+    private Map<String, String> xmlToMetacard;
+
+    private static List<String> watchedElements = new ArrayList<>();
+
+    static {
+        watchedElements.add("urn:catalog:metacard");
+    }
+
+    @Override
+    public List<Attribute> getAttributes() {
+        return attributes;
+    }
+
+    @Override
+    public List<String> getWatchedElements() {
+        return null;
+    }
+
+    @Override
+    public void setDocumentLocator(Locator locator) {
+
+    }
+
+    protected XMLSaxEventHandlerImpl() {
+        xmlToMetacard = new HashMap<>();
+        xmlToMetacard.put("title", Metacard.TITLE);
+        xmlToMetacard.put("point-of-contact", Metacard.POINT_OF_CONTACT);
+        xmlToMetacard.put("description", Metacard.DESCRIPTION);
+        xmlToMetacard.put("source", Metacard.RESOURCE_URI);
```

Comment:
They look like the same values, so really could use the constants for both sides. Would be slightly unclear so might deserve a comment, but more correct than duplicating the string values of those constants, I think.